inference-engine/src/inference_engine/ie_core.cpp

```diff
@@ -158,6 +168,46 @@ class Core::Impl : public ICore {
 
     mutable std::map<std::string, InferencePlugin> plugins;
 
+    class CoreConfig final {
+    public:
+        struct CacheConfig {
+            std::shared_ptr<ICacheManager> _cacheManager;
+        };
+        CoreConfig() = default;
+
+        void setAndUpdate(std::map<std::string, std::string>& config) {
+            std::lock_guard<std::mutex> lock(_cacheConfigMutex);
```

Comment:
should not we move this lock after
```
if (it != config.end()) {
```
In this case mutex is locked only if cache dir is passed, all other cases happen w/o mutex lock